Docs: Mobile docs side navigation toggle

garrying opened the pull request:
## What this PR does / why we need it
This introduces a button to toggle side navigation visibility in `/docs` at smaller viewports. The button and side navigation toggling is only applied to the `small` breakpoint.

| Before      | After |
| ----------- | ----------- |
| ![tanzucommunityedition io_docs_latest_(iPhone XR)](https://user-images.githubusercontent.com/227587/149583358-6cd1b0bd-1f66-4b93-b294-540a2e3d3f4b.png) | ![localhost_1313_docs_latest_(iPhone XR)](https://user-images.githubusercontent.com/227587/149583372-8d94e3fc-45a9-46f5-a656-0d231dbfed4c.png) |


## Details for the Release Notes (PLEASE PROVIDE)
<!--
Unless this is a trivial change, we want to know more about your contribution!
This can even be a TLDR version of the "What this PR does".
If a trivial change, just write "NONE" in the release-note block below.
Otherwise, a release note is required:
-->
```release-note

```

## Which issue(s) this PR fixes
Fixes: #2119

## Describe testing done for PR
- `hugo server`
- Navigate to `/docs`
- Reduce viewport width to smaller than `767px` or visit on a mobile browser

## Special notes for your reviewer
The button label is currently `Table of contents`. Open to suggestions on different label name and styling.

joshrosso reviewed (changes requested):
Looks great @garrying.

Can you squash the commits in this PR to 1 commit with a detailed commit message?

Then we'll merge. Thank you!

garrying committed:
Improve mobile navigation for docs

- Adding a toggle button element to toggle visibility of the
side navigation.
- Hide side navigation on smaller viewports.

Signed-off-by: Garry Ing <ging@vmware.com>
